src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/DocumentHighlighting/DocumentHighlightEndpointTest.cs

```diff
@@ -0,0 +1,193 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+#nullable disable
+
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor.LanguageServer.Common;
+using Microsoft.AspNetCore.Razor.LanguageServer.EndpointContracts;
+using Microsoft.AspNetCore.Razor.LanguageServer.Extensions;
+using Microsoft.AspNetCore.Razor.LanguageServer.Protocol;
+using Microsoft.AspNetCore.Razor.LanguageServer.Test;
+using Microsoft.AspNetCore.Razor.LanguageServer.Test.Common;
+using Microsoft.AspNetCore.Razor.Test.Common;
+using Microsoft.CodeAnalysis.Razor.Workspaces;
+using Microsoft.CodeAnalysis.Razor.Workspaces.Extensions;
+using Microsoft.CodeAnalysis.Testing;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Moq;
+using OmniSharp.Extensions.JsonRpc;
+using OmniSharp.Extensions.LanguageServer.Protocol.Server;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.DocumentHighlighting.Test
+{
+    [UseExportProvider]
+    public class DocumentHighlightEndpointTest : LanguageServerTestBase
+    {
+        [Fact]
+        public async Task Handle_SingleServer_Local()
```

Comment:
Sadly there are no tests for HTML, because we can't use the real HTML language server. That means we'd end up with code somewhere around line 178 on this file that does `return 4`, and then an assert somewhere that validates that we got back a 4. I find that type of test code to be useless, so I didn't write it.

I might try and follow up with an integration test for highlighting though (and we should probably do one for definition, hover and rename too, if we don't have them already...)